### PR TITLE
XWIKI-15444: Export UI does not export child pages if there are many (and tree has pagination)

### PR DESCRIPTION
**Pull request by surli**
## Issue

https://jira.xwiki.org/browse/XWIKI-15444

## Solution

  * Form is now created on the fly to avoid nested form

## Tests

All dedicated tests running.

### CODE REVIEW

**Comment by surli** [comment]
@tmortagne @mflorea this is ready.

**Comment by surli** [comment]
Edit: it's not ready yet, I forgot to edit the pageobjects for the tests.

**Comment by surli** [comment]
ok this is ready @mflorea @tmortagne

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/export_modal.vm [hunk 1]
```diff
@@ -59,7 +59,7 @@
       </div>
     </div>
     <div id="exportModelOtherCollapse" class="panel-collapse collapse" role="tabpanel" aria-labelledby="exportModalHeadingOther">
-      <form id="export-form" method="post" class="panel-body">
+      <div id="export-div" class="panel-body">
```
[the remark this answers is not in the document]
Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/export_modal.vm]: I actually use the div in my functional tests. And about `exportModelOtherCollapse` I did not take the risk of renaming it :)

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/exporter/exporter.js [hunk 2]
```diff
@@ -305,9 +303,17 @@ require([
...
           };
 
-          // we clean the form from old hidden pages (in case we click several times on the export button)
-          $("input[name='pages']").remove();
-          $("input[name='excludes']").remove();
+          // in case an old form was already existing
+          // we could remove it after the submit, but it's easier that way for testing
+          $('#export-form').remove();
+
+          $('<form id="export-form" />').appendTo('#export-div');
```
[the remark this answers is not in the document]
Comment (surli) [reply xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/exporter/exporter.js]: I assume I could put it to the body if you think it's cleaner.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/export_modal.vm (hunk 1 above)
[the remark this answers is not in the document]
Comment (mflorea) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/export_modal.vm]: To be clear, my suggestion is to keep ``<div class="panel-body">`` (without the id attribute) and to use the parent ``exportModelOtherCollapse`` in tests and JavaScript if it's not too complicated (and indeed, renaming the id might require other changes).

[the remark this answers is not in the document]
Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/export_modal.vm]: yeah I got that, applying the change right now :)

**Comment by surli** [comment]
Just applied your suggestions @mflorea functional tests are still passing.

**Pull request merged**